tools/ktx/encode_utils.h

```diff
@@ -0,0 +1,384 @@
+// Copyright 2022-2023 The Khronos Group Inc.
+// Copyright 2022-2023 RasterGrid Kft.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "command.h"
+#include "utility.h"
+
+#include <thread>
+#include <unordered_map>
+
+// -------------------------------------------------------------------------------------------------
+
+namespace ktx {
+
+enum class EncodeCodec {
+    NONE = 0,
+    BasisLZ,
+    UASTC,
+    INVALID = 0x7FFFFFFF
+};
+
+template <bool ENCODE_CMD>
+struct OptionsCodec {
```

Comment:
This is coming from the project specification: `--codec: basis-lz | uastc`. Only supercompressions are allowed, not specific GPU block compressions.

Reply:
There is confusion here about "supercompression". Supercompression is BasisLZ, zstd and zlib. The BasisU encoders primarily encode to a block compression scheme (ETC1S or UASTC) and then in the case of ETC1S apply BasisLZ supercompression.

So fundamentally what is happening here is no different than ASTC compression (and BC7 in the future). To anyone who understands this, it is completely confusing to have ASTC encoding done in `create` and BasisU done in `encode`.

From a user perspective it must be possible to create files with any of the 3 supported block-compressed encoding schemes and supercompression applied and to encode .ktx2 files to any scheme that is not incompatible with its current format and apply supercompression.

The current `ktxsc` tool is misnamed for historical reasons. In functionality it can encode to UASTC, ASTC and BasisLZ. It can also apply zstd supercompression to anything but BasisLZ format files.